infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -578,15 +581,18 @@ private void onStandbyClusterAddRemove(TopologyDescriptor discoveredTopology) {
         log.debug("Standby Cluster has been added or removed from topology={}", discoveredTopology);
 
         // We only need to process new standby's if your role is of an ACTIVE cluster
-        if (localClusterDescriptor.getRole() == ClusterRole.STANDBY) {
-            return;
-        }
-
-        if (replicationManager != null && isLeader.get()) {
-            replicationManager.processStandbyChange(discoveredTopology);
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
+            if (replicationManager != null && isLeader.get()) {
+                replicationManager.processStandbyChange(discoveredTopology);
+            }
         }
 
         updateLocalTopology(discoveredTopology);
+        updateReplicationManagerTopology(discoveredTopology);
+        // Update topology config id in metadata manager
+        logReplicationMetadataManager.setupTopologyConfigId(topologyDescriptor.getTopologyConfigId());
+        log.debug("Persist new topologyConfigId {}, cluster id={}, status={}", topologyDescriptor.getTopologyConfigId(),
+                localClusterDescriptor.getClusterId(), localClusterDescriptor.getRole());
```

Comment:
Even if current node's role did not change, it does not guarantee that others' role or config id is not changed...  
So updating sender/receiver's config id is necessary here.